src/blk/kernel/KernelDevice.h

```diff
@@ -27,6 +27,10 @@
 
 #define RW_IO_MAX (INT_MAX & CEPH_PAGE_MASK)
 
+enum struct blk_access_mode_t {
```

Comment:
i am wondering if this enum is used other where than `KernelDevice.cc`, if not, how about moving it into `KernelDevice.cc`?

Reply:
@tchaikov 
I'm currently only changing the code that I need to get BlueStore ready on FreeBSD, but there is another part in the `blk`tree that perhaps deserves an overhoal:
```
./src/blk/zoned/HMSMRDevice.h:35:  std::vector<int> fd_directs, fd_buffereds;
./src/blk/zoned/HMSMRDevice.cc:61:  fd_buffereds.resize(WRITE_LIFE_MAX, -1);
./src/blk/zoned/HMSMRDevice.cc:145:    fd_buffereds[i] = fd;
./src/blk/zoned/HMSMRDevice.cc:164:    if (fcntl(fd_buffereds[i], F_SET_FILE_RW_HINT, &i) < 0) {
./src/blk/zoned/HMSMRDevice.cc:183:  r = posix_fadvise(fd_buffereds[WRITE_LIFE_NOT_SET], 0, 0, POSIX_FADV_RANDOM);
./src/blk/zoned/HMSMRDevice.cc:221:    BlkDev blkdev_buffered(fd_buffereds[WRITE_LIFE_NOT_SET]);
./src/blk/zoned/HMSMRDevice.cc:277:    if (fd_buffereds[i] >= 0) {
./src/blk/zoned/HMSMRDevice.cc:278:      VOID_TEMP_FAILURE_RETRY(::close(fd_buffereds[i]));
./src/blk/zoned/HMSMRDevice.cc:279:      fd_buffereds[i] = -1;
./src/blk/zoned/HMSMRDevice.cc:312:    assert(fd_buffereds[i] >= 0);
./src/blk/zoned/HMSMRDevice.cc:313:    VOID_TEMP_FAILURE_RETRY(::close(fd_buffereds[i]));
./src/blk/zoned/HMSMRDevice.cc:314:    fd_buffereds[i] = -1;
./src/blk/zoned/HMSMRDevice.cc:355:  int r = ::fstat(fd_buffereds[WRITE_LIFE_NOT_SET], &st);
./src/blk/zoned/HMSMRDevice.cc:362:    BlkDev blkdev{fd_buffereds[WRITE_LIFE_NOT_SET]};
./src/blk/zoned/HMSMRDevice.cc:428:  return buffered ? fd_buffereds[write_hint] : fd_directs[write_hint];
./src/blk/zoned/HMSMRDevice.cc:859:    r = ::sync_file_range(fd_buffereds[WRITE_LIFE_NOT_SET], off, len, SYNC_FILE_RANGE_WRITE|SYNC_FILE_RANGE_WAIT_AFTER|SYNC_FILE_RANGE_WAIT_BEFORE);
./src/blk/zoned/HMSMRDevice.cc:1026:  int r = ::pread(buffered ? fd_buffereds[WRITE_LIFE_NOT_SET] : fd_directs[WRITE_LIFE_NOT_SET],
./src/blk/zoned/HMSMRDevice.cc:1154:      r = ::pread(fd_buffereds[WRITE_LIFE_NOT_SET], t, left, off);
./src/blk/zoned/HMSMRDevice.cc:1208:  int r = posix_fadvise(fd_buffereds[WRITE_LIFE_NOT_SET], off, len, POSIX_FADV_DONTNEED);
```

Very likely the `HMSMRDevice` code was writen/derived from the `KernelDevice` code.
As such it would even warant to migrate a lot of the common needed code up into `src/blk/BlockDevice.*`
But perhaps for now that is a too big a step.

Reply:
@tchaikov 
Because of this, I did move definitions and functions into `src/blk/BlockDevice.*`